l5kit/l5kit/data/map_api.py

```diff
@@ -200,6 +311,66 @@ def is_traffic_face_colour(self, element_id: str, colour: str) -> bool:
             return True
         return False
 
+    @lru_cache(maxsize=CACHE_SIZE)
+    def get_color_for_face(self, face_id: str) -> str:
+        """
+        Utility function. It calls `is_traffic_face_colour` for a set of colours until it gets an answer.
+        If no color is found, then `face_id` is not the id of a traffic light face (and we raise ValueError).
+
+        Args:
+            face_id (str): the element id
+        Returns:
+            str: the colour as string for this traffic face
+        """
+        for color in self.TR_FACES_COLORS:
+            if self.is_traffic_face_colour(face_id, color):
+                return color
+        raise ValueError(f"Face {face_id} has no valid color among {self.TR_FACES_COLORS}")
+
+    def get_bounds(self) -> dict:
+        """
+        For each elements of interest returns bounds [[min_x, min_y],[max_x, max_y]] and proto ids
+        Coords are computed by the MapAPI and, as such, are in the world ref system.
+
+        Returns:
+            dict: keys are classes of elements, values are dict with `bounds` and `ids` keys
+        """
+        lanes_ids = []
+        crosswalks_ids = []
+
+        lanes_bounds = np.empty((0, 2, 2), dtype=np.float)  # [(X_MIN, Y_MIN), (X_MAX, Y_MAX)]
+        crosswalks_bounds = np.empty((0, 2, 2), dtype=np.float)  # [(X_MIN, Y_MIN), (X_MAX, Y_MAX)]
+
+        for element in self.elements:
+            element_id = MapAPI.id_as_str(element.id)
+
+            if self.is_lane(element):
+                lane = self.get_lane_coords(element_id)
+                x_min = min(np.min(lane["xyz_left"][:, 0]), np.min(lane["xyz_right"][:, 0]))
+                y_min = min(np.min(lane["xyz_left"][:, 1]), np.min(lane["xyz_right"][:, 1]))
+                x_max = max(np.max(lane["xyz_left"][:, 0]), np.max(lane["xyz_right"][:, 0]))
+                y_max = max(np.max(lane["xyz_left"][:, 1]), np.max(lane["xyz_right"][:, 1]))
+
+                lanes_bounds = np.append(lanes_bounds, np.asarray([[[x_min, y_min], [x_max, y_max]]]), axis=0)
+                lanes_ids.append(element_id)
+
+            if self.is_crosswalk(element):
+                crosswalk = self.get_crosswalk_coords(element_id)
+                x_min = np.min(crosswalk["xyz"][:, 0])
+                y_min = np.min(crosswalk["xyz"][:, 1])
```

Comment:
I think you can reduce the calls here to have like `x_min, y_min = np.min(crosswalk["xyz"], axis=0)`